exchanges/orderbook/tranches.go

```diff
@@ -78,27 +78,23 @@ updates:
 
 // deleteByID deletes reference by ID
 func (ts *Tranches) deleteByID(updts Tranches, bypassErr bool) error {
-updates:
 	for x := range updts {
-		for y := range *ts {
-			if updts[x].ID != (*ts)[y].ID {
-				continue
-			}
+		idx := slices.IndexFunc(*ts, func(t Tranche) bool {
+			return t.ID == updts[x].ID
+		})
 
-			if y < len(*ts) {
-				copy((*ts)[y:], (*ts)[y+1:])
-				*ts = (*ts)[:len(*ts)-1]
-			} else {
-				*ts = slices.Delete(*ts, y, y+1)
+		if idx == -1 {
+			if !bypassErr {
+				return fmt.Errorf("delete error: %w %d not found", errIDCannotBeMatched, updts[x].ID)
 			}
-			continue updates
-		}
-		if !bypassErr {
-			return fmt.Errorf("delete error: %w %d not found",
-				errIDCannotBeMatched,
-				updts[x].ID)
+			continue
 		}
+
+		*ts = slices.Delete(*ts, idx, idx+1)
 	}
+
+	*ts = slices.Clip(*ts)
```

Comment:
We don't need to clip as the underlying array is fine to stay big, every time a new update comes in it will resize the underlying and hurt performance.

Reply:
I'm wary about this rationale; Doesn't it pin us to high-watermark memory usage?
If there's an absurd update, and the orderbook shrinks by 75% ... would we reclaim that?

Reply:
On very liquid pairs we'd likely not see an orderbook shrink by 75%, esp since most exchanges will have a limit of N for ob depth on websocket which continually refresh at that limit but could potentially happen on illiquid pairs and they generally update way less, so in that case perf won't be as critical.

Perhaps a conditional thing? 

```
if cap(*ts) > 2*len(*ts) {
	*ts = slices.Clip(*ts)
}
```

But this adds complexity and we'd need hard data to prove either case